### PR TITLE
test(species): add comprehensive tests for helper functions and apply modern Go features

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

- Add 37 comprehensive test cases for species tracker helper functions, improving coverage from 95.0% to 96.7%
- Apply modern Go 1.22+ features (maps.Keys, slices.Collect, min/max built-ins)
- Remove unnecessary runtime guards and reduce cognitive complexity in tests

### New Tests (`species_tracker_helper_functions_test.go` - 813 lines)

| Test Suite | Coverage Target | Test Cases |
|------------|-----------------|------------|
| `TestCheckAndUpdateLifetimeLocked_EdgeCases` | update.go:111 | 8 cases: boundary conditions, nanosecond precision, negative days anomaly |
| `TestLoadNotificationHistoryFromDatabase_EdgeCases` | database.go:294 | 6 cases: type filtering, duplicate handling, nil map init, error paths |
| `TestPruneYearlyEntriesLocked_CustomResetDates` | maintenance.go:90 | 6 cases: July 1/Dec 31 fiscal years, before/after reset boundaries |
| `TestIsInEarlyWinterMonths_HemisphereVariations` | season.go:89 | 13 cases: Northern/Southern hemisphere, edge months |
| `TestLoadSingleSeasonData_ErrorPaths` | database.go:203 | 5 cases: invalid dates, empty results, database errors |
| `TestCalculateDaysSince_EdgeCases` | status.go:110 | 5 cases: same time, negative days, large differences |
| `TestAllSeasonsEmpty` | database.go:253 | 5 cases: nil/empty maps, partial data |

### Code Quality Improvements

- **Modern Go features**: Use `maps.Keys()` + `slices.Collect()` instead of manual iteration; use `min()`/`max()` built-ins
- **Simplified testing.go**: Remove `testing.Testing()` runtime guards - methods are clearly documented as test-only
- **Reduced cognitive complexity**: Extract helper functions in tests to stay below gocognit threshold (12)

## Coverage Improvement

| Function | Before | After |
|----------|--------|-------|
| `loadSingleSeasonData` | 81.2% | 100% |
| `loadNotificationHistoryFromDatabase` | 64.7% | 100% |
| `pruneYearlyEntriesLocked` | 91.7% | 100% |
| `isInEarlyWinterMonths` | 85.7% | 100% |
| **Overall package** | 95.0% | 96.7% |

## Test plan

- [x] All 37 new test cases pass
- [x] Full package tests pass with race detector: `go test -race ./internal/analysis/species/...`
- [x] Linter passes: `golangci-lint run internal/analysis/species/`
- [x] No production bugs discovered - edge cases are handled correctly

## Summary by CodeRabbit

* **Chores**
  * Enhanced asynchronous operation tracking and shutdown synchronization for more reliable cleanup.
  * Improved debug logging performance by conditionally formatting messages only when debug logging is enabled.
  * Added runtime safety guards to prevent test-only methods from being used in production code.

* **Tests**
  * Added comprehensive unit tests covering edge cases for helper functions and error handling paths.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: internal/analysis/species/maintenance.go [hunk 1]
```diff
@@ -251,7 +251,7 @@ func (t *SpeciesTracker) RecordNotificationSent(scientificName string, sentTime
 	// If a goroutine does leak due to database hang, in-memory suppression still works.
 	// TODO(BG-17): Consider adding context.Context parameter to SaveNotificationHistory interface
 	if t.ds != nil {
-		go func() {
+		t.asyncOpsWg.Go(func() {
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/analysis/species/maintenance.go:254]: You are incorrect, this is a valid go 1.25 pattern

File: internal/analysis/species/maintenance.go [hunk 2]
```diff
@@ -307,7 +307,7 @@ func (t *SpeciesTracker) CleanupOldNotificationRecords(currentTime time.Time) in
 	// However, SQLite is local and GORM has internal timeouts, so hangs are unlikely.
 	// TODO(BG-17): Consider adding context.Context parameter to DeleteExpiredNotificationHistory interface
 	if t.ds != nil {
-		go func() {
+		t.asyncOpsWg.Go(func() {
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/analysis/species/maintenance.go:310]: You are incorrect, this is a valid go 1.25 pattern
